### PR TITLE
Add factory constructor Multimap.fromMultimap

### LINKED ISSUE #705: Implement a factory constructor on Multimap that copies another Multimap
Map has .from() and .of() factory constructors, why doesn't Multimap?

### PR DESCRIPTION
**Pull request by patrikbadanjak**
A little convenience method for easier copying. Closes #705.

### CODE REVIEW

File: lib/src/collection/multimap.dart [hunk 1]
```diff
@@ -30,6 +30,10 @@ abstract class Multimap<K, V> {
       {K Function(dynamic)? key,
       V Function(dynamic)? value}) = ListMultimap<K, V>.fromIterable;
 
+  /// Constructs a new multimap with the same values as [other]
```
Comment (cbracken) [inline lib/src/collection/multimap.dart:33]: Please add a period at the end of the doc comment.

Ref: https://dart.dev/guides/language/effective-dart/documentation

File: lib/src/collection/multimap.dart [hunk 2]
```diff
@@ -119,6 +123,11 @@ abstract class _BaseMultimap<K, V, C extends Iterable<V>>
     }
   }
 
+  /// Constructs a new multimap with the same values as [other]
```
Comment (cbracken) [inline lib/src/collection/multimap.dart:126]: Similar comment.

File: lib/src/collection/multimap.dart [hunk 3]
```diff
@@ -240,6 +249,9 @@ class ListMultimap<K, V> extends _BaseMultimap<K, V, List<V>> {
   ListMultimap.fromIterable(super.iterable, {super.key, super.value})
       : super.fromIterable();
 
+  /// Constructs a new multimap with the same values as [other]
```
Comment (cbracken) [inline lib/src/collection/multimap.dart:252]: And again here.

File: test/collection/multimap_test.dart [hunk 4]
```diff
@@ -64,6 +64,41 @@ void main() {
...
+
+    test('should have same values', () {
+      var map1ContainsAllMap2Values =
+          mmap1.values.every((value) => mmap2.containsValue(value));
+      var map2ContainsAllMap1Values =
+          mmap2.values.every((value) => mmap1.containsValue(value));
+
+      expect(true, map1ContainsAllMap2Values && map2ContainsAllMap1Values);
+    });
+  });
```
Comment (cbracken) [inline test/collection/multimap_test.dart:100]: We should add a test to ensure no one inadvertently breaks the behaviour by, say, re-using the Lists/Sets we create internally in mmap1 -- either by modifying the factory constructor or by modifying addAll. I haven't looked to see that we already have this test for addAll (hopefully), but either way, it's worth being paranoid here.

Something like:
```dart
mmap1 = Multimap()
  ..add('k1', 'v1')
  ..add('k1', 'v2')
  ..add('k2', 'v3');
mmap2 = Multimap.fromMultimap(mmap1);
mmap1['k1'].remove('v1');
expect(mmap2['k1'].contains('v1'));
```

**Review by cbracken — COMMENTED** [review]
Thanks for your patch! Just a few comments.

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

**Comment by patrikbadanjak** [comment]
I right clicked on the files in Android Studio and clicked on 'reformat code with dart format'. Then I ran it from the terminal and it produced a different output...

**Comment by cbracken** [comment]
Interesting -- sounds like maybe a bug in the intellij/android studio integration. If it's reproducible, you could file a bug against https://github.com/dart-lang/sdk.

**Comment by cbracken** [comment]
Thanks again for the patch!

**Comment by patrikbadanjak** [comment]
The terminal uses the homebrew version unlike Android Studio which uses the one from the Flutter package. They have the same version code.

**Comment by cbracken** [comment]
Ah - I'm using the commandline version from the SDK included in Flutter and it matches the corrected format but I'll bet that's because my flutter install is always near tip-of-tree on master branch since working on flutter is my day job. We did some significant updates for Dart 3 between the last stable release and now so perhaps that accounts for the difference.

Thanks for the additional info!

**Pull request merged**